tools/cellprofiler/common.xml

```diff
@@ -0,0 +1,445 @@
+<tool id="cp_common" name="Cellprofiler common start modules" version="@CP_VERSION@">
+    <description>
+        The first 4 modules in Cellprofiler to start a pipeline, namely Images,Metadata, NameAndTypes, Groups 
+    </description>
+    <macros>
+        <import>macros.xml</import>
+        <xml name="image_type_condition"> 
+            <conditional name="con_set_intensity">
+                <param name="set_intensity_range_from" type="select" label="Set intensity range from">
+                    <option value="Image metadata">Image metadata</option>
+                    <option value="Image bit-depth">Image bit-depth</option>
+                    <option value="Manual">Manual</option>
+                </param>
+                    <when value="Manual">
+                        <param name="maximum_intensity" type="float" value="255.0" label="Maximum intensity"/>
+                    </when>
+                    <when value="Image metadata">
+                    </when>
+                    <when value="Image bit-depth">
+                    </when>
+            </conditional>
+        </xml>
+        <xml name="name_type_rule_matching_file">
+            <param name="operator" type="select">
+                <option value="does">Does</option>
+                <option value="doesnot">Doesnot</option>
+            </param>
+            <param name="contain" type="select">
+                <option value="contain">Contain</option>
+                <option value="Contain regular expression">Contain regular expression</option>
+                <option value="startwith">Start with</option>
+                <option value="endwith">End with</option>
+                <option value="Exactly match">Exactly match</option>
+            </param>
+            <param name="match_value" type="text"/>
+        </xml>
+        <xml name="image_matching_rules">
+            <repeat name="r_match_rule" title="Rules">
+                <param name="match_all_any" type="select" display="radio" label="Match the following rules">
+                    <option value="and">All</option>
+                    <option value="or">Any</option>
+                </param>
+                <conditional name="con_match">
+                      <param name="rule_type" type="select" label="Select rule criteria">
+                          <option value="file">File</option>
+                          <option value="directory">Directory</option>

```

Comment:
does directory make sense in our use-case?

Reply:
It's just what is shown in Cellprofiler....maybe not for our use case but it is an option in cellprofiler